### ISSUE TITLE
Signed and unsigned values not matching in switch()

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#3044]
Repro is with Zed commit bcb1697. This issue was found while helping a community user craft flow control logic for a Zed shaper script.

Start with this CSV data:

```
$ cat data.csv 
event_id
3
4
```

When read in, these values are treated as floats, as indicated by the `.` in the values in the ZSON output.

```
$ zq -version
Version: v0.29.0-512-gbcb16977

$ zq -z -i csv data.csv | tee float-data.zson
{event_id:3.}
{event_id:4.}
```

The user's _first_ problem was that these values didn't match in their `switch()` statement.

```
$ cat switch.zed 
switch event_id (
  3 => put note := "I saw a three";
  4 => put note := "I saw a four";
)

$ zq -I switch.zed float-data.zson
[no output]
```

I wasn't sure if that was supposed to work since I know float/integer equality is not a given in some languages. I thought some casting during preprocessing would do the trick as a workaround, but that didn't work either.

```
$ zq -z -i csv 'put event_id:=uint64(event_id)' data.csv | tee uint64-data.zson 
{event_id:3(uint64)}
{event_id:4(uint64)}

$ zq -I switch.zed uint64-data.zson 
[no output]
```

It's only when I casted to a _signed_ int that it started working.

```
$ zq -z -i csv 'put event_id:=int64(event_id)' data.csv | tee int64-data.zson 
{event_id:3}
{event_id:4}

$ zq -I switch.zed int64-data.zson 
{event_id:4,note:"I saw a four"}
{event_id:3,note:"I saw a three"}
```

In conclusion, I'm almost certain the failed quality test between signed/unsigned is a bug. I'm less sure of the float one.

**Comment by philrz** [comment]
I happened to revisit this one in an issue scrub and have confirmed that it's still with us as of super commit ed23b7a. Here's an update of the repro aligned with current formats and `switch` syntax.

```
$ cat float-data.sup 
{event_id:3.}
{event_id:4.}

$ cat int64-data.sup 
{event_id:3}
{event_id:4}

$ cat uint64-data.sup 
{event_id:3::uint64}
{event_id:4::uint64}

$ cat switch.spq 
switch event_id
  case 3 ( put note := "I saw a three" )
  case 4 ( put note := "I saw a four" )

$ super -version
Version: v0.2.0-13-ged23b7a48

$ super -I switch.spq float-data.sup 
[no output]

$ super -I switch.spq uint64-data.sup 
[no output]

$ super -I switch.spq int64-data.sup 
{event_id:3,note:"I saw a three"}
{event_id:4,note:"I saw a four"}
```
